STEP/CodeGeneration/CodeGenerationVisitor.cs

```diff
@@ -506,15 +527,18 @@ public void Visit(FuncsNode n)
         }
     }
 
-    public void Visit(RetNode n)
-    {
+    public void Visit(RetNode n) {
         // If no expression, emit empty return
-        if (n.RetVal is null)
-        {
+        if (n.RetVal is null) {
             EmitLine("return;");
         }
-        else
-        {
+        else {
+            // If returning from function, free arrays declared in scope
+            if (n.SurroundingFuncType.IsArray) {
+                // If in outer scope, free all arrays except returned and remove from list - else free without removing from list
+                bool isInOuterScope = n.Type.ScopeLevel - 1 == n.SurroundingFuncType.ScopeLevel;
+                FreeArrays(n.SurroundingFuncType.ScopeLevel, (IdNode) n.RetVal, isInOuterScope);
```

Comment:
Forstår ikke helt det her. isInOuterScope er vel true når at return ikke står inde i en if-stmt for eksempel right? Altså hvis return bare står som en stmt.

Reply:
Hvorfor er det så vi skal fjerne dem fra listen når den er i outer scope men ikke skal fjerne dem når den ikke er i outer scope?